cypress-smoketests/integration/search.spec.js

```
@@ -0,0 +1,92 @@
+  searchByPractitionerName,
+  searchOpinionByKeyword,
+} = require('../support/pages/advanced-search');
+const { goToCaseDetail } = require('../support/pages/case-detail');
+
+const { getUserToken, login } = require('../support/pages/login');
+
+const docketNumberToSearchBy = '103-20';
+const barNumberToSearchBy = 'PT1234';
+let token;
+
+describe('Case Advanced Search', () => {
+  before(async () => {
+    const results = await getUserToken(
+      'docketclerk1@example.com',
+      'Testing1234$',
+    );
+    token = results.AuthenticationResult.IdToken;
+  });
+
+  it('should be able to login', () => {
+    login(token);
+  });
+
+  it('should be able to search for case by practitioner name', () => {
+    gotoAdvancedSearch();
+    searchByPetitionerName();
+  });
+
+  it('should be able to search for case by docket number', () => {
+    gotoAdvancedSearch();
+    searchByDocketNumber(docketNumberToSearchBy);
+  });
+});
+
+describe('Practitoner Search', () => {
+  before(async () => {
+    const results = await getUserToken(
+      'docketclerk1@example.com',
+      'Testing1234$',
+    );
+    token = results.AuthenticationResult.IdToken;
+  });
+
+  it('should be able to login', () => {
+    login(token);
+  });
+
+  it('should be able to search for a practitioner by name', () => {
+    gotoAdvancedPractitionerSearch();
+    searchByPractitionerName();
+  });
+
+  it('should be able to search for for a practitioner by bar number', () => {
+    gotoAdvancedPractitionerSearch();
+    searchByPractitionerbarNumber(barNumberToSearchBy);
+  });
+});
+
+describe('Opinion Search', () => {
+  before(async () => {
+    const results = await getUserToken(
+      'docketclerk1@example.com',
+      'Testing1234$',
+    );
+    token = results.AuthenticationResult.IdToken;
+  });
+
+  it('should be able to login', () => {
+    login(token);
+  });
+
+  it('should create an opinion to search for', () => {
+    goToCaseDetail(docketNumberToSearchBy);
+    createOpinion();
```

This is going to add a new opinion to the same case each time, and I think these could actually silently fail, because there will be old data there to search for, right?

---

Hmm...that is a really good point. However, I can't assume that there would be other cases on the env apart from the one I am planning to create in this test (based on your feedback). Do you think a better approach would be to add orders and opinions as a part of other case creation related smoketests and then expect those docs to show up in the search results?

---

Could you just add the opinion to the case you're going to create as part of this test? I think that should work. You should be able to check the results for the docket number of the case you created.